### DISCUSSION TITLE
TinyMCE "Source Code" Editor Restrictions

### DISCUSSION THREAD

**Discussion by Amir-Zouerami** [pocketbase/pocketbase#3773] [Q&A] [answered]
Thanks to the TinyMCE implementation I write my blog posts directly in the pocketbase admin panel. The problem is TinyMCE does not allow me to add an arbitrary class to the \<code> element through the "source code" editor button. I have my own syntax highlighting system for my blog and it requires me to set a `language` prop. I use this class to insert what language the code snippet is in.

The problem is TinyMCE does not allow me to manually add this but has no problem with me adding classes to other elements. When I do, it automatically removes the class attribute from \<code>. Anything that allows me to override this?

**Comment by ganigeorgiev** [comment, marked as answer]
Hm, by looking at the TinyMCE [Code plugin docs](https://www.tiny.cloud/docs/tinymce/6/codesample/) there are no options to control this behavior.

The only way I was able to kindof add a new class to the `code` element was to insert the raw code snippet directly through the "Source code" editor (_the second button from the right_).
Removing the `language-*` class from the parent `<pre>` that is added automatically also seems to allow  customizing the `code` element behavior but similar to the direct code input it is no longer recognized as "Code sample" entry and need to be edited manually.

**Reply by Amir-Zouerami** [reply]
I was afraid that would be the case. Good enough I guess. Thank you.
